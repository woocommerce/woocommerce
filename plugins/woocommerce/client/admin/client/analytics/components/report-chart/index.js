/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { Component } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { format as formatDate } from '@wordpress/date';
import { withSelect } from '@wordpress/data';
import { get, isEqual } from 'lodash';
import PropTypes from 'prop-types';
import { Chart, AnalyticsError } from '@woocommerce/components';
import {
	getReportChartData,
	getTooltipValueFormat,
	SETTINGS_STORE_NAME,
	REPORTS_STORE_NAME,
} from '@woocommerce/data';
import {
	getAllowedIntervalsForQuery,
	getCurrentDates,
	getDateFormatsForInterval,
	getIntervalForQuery,
	getChartTypeForQuery,
} from '@woocommerce/date';
import { CurrencyContext } from '@woocommerce/currency';

/**
 * Internal dependencies
 */
import {
	getChartMode,
	getSelectedFilter,
	createDateFormatter,
	buildChartData,
} from './utils';

/**
 * Component that renders the chart in reports.
 */
export class ReportChart extends Component {
	shouldComponentUpdate( nextProps ) {
		if (
			nextProps.isRequesting !== this.props.isRequesting ||
			nextProps.primaryData.isRequesting !==
				this.props.primaryData.isRequesting ||
			nextProps.secondaryData.isRequesting !==
				this.props.secondaryData.isRequesting ||
			! isEqual( nextProps.query, this.props.query )
		) {
			return true;
		}

		return false;
	}

	getItemChartData() {
		const { primaryData, selectedChart } = this.props;
		const chartData = primaryData.data.intervals.map( function (
			interval
		) {
			const intervalData = {};
			interval.subtotals.segments.forEach( function ( segment ) {
				if ( segment.segment_label ) {
					const label = intervalData[ segment.segment_label ]
						? segment.segment_label +
						  ' (#' +
						  segment.segment_id +
						  ')'
						: segment.segment_label;
					intervalData[ segment.segment_id ] = {
						label,
						value: segment.subtotals[ selectedChart.key ] || 0,
					};
				}
			} );
			return {
				date: formatDate( 'Y-m-d\\TH:i:s', interval.date_start ),
				...intervalData,
			};
		} );
		return chartData;
	}

	getTimeChartData() {
		const {
			query,
			primaryData,
			secondaryData,
			selectedChart,
			defaultDateRange,
		} = this.props;
		const currentInterval = getIntervalForQuery( query, defaultDateRange );
		const { primary, secondary } = getCurrentDates(
			query,
			defaultDateRange
		);

		return buildChartData(
			primaryData,
			secondaryData,
			primary,
			secondary,
			query.compare,
			selectedChart.key,
			currentInterval
		);
	}

	getTimeChartTotals() {
		const { primaryData, secondaryData, selectedChart } = this.props;

		return {
			primary: get(
				primaryData,
				[ 'data', 'totals', selectedChart.key ],
				null
			),
			secondary: get(
				secondaryData,
				[ 'data', 'totals', selectedChart.key ],
				null
			),
		};
	}

	renderChart( mode, isRequesting, chartData, legendTotals ) {
		const {
			emptySearchResults,
			filterParam,
			interactiveLegend,
			itemsLabel,
			legendPosition,
			path,
			query,
			selectedChart,
			showHeaderControls,
			primaryData,
			defaultDateRange,
		} = this.props;
		const currentInterval = getIntervalForQuery( query, defaultDateRange );
		const allowedIntervals = getAllowedIntervalsForQuery(
			query,
			defaultDateRange
		);
		const formats = getDateFormatsForInterval(
			currentInterval,
			primaryData.data.intervals.length,
			{ type: 'php' }
		);
		const emptyMessage = emptySearchResults
			? __( 'No data for the current search', 'woocommerce' )
			: __( 'No data for the selected date range', 'woocommerce' );
		const { formatAmount, getCurrencyConfig } = this.context;
		return (
			<Chart
				allowedIntervals={ allowedIntervals }
				data={ chartData }
				dateParser={ '%Y-%m-%dT%H:%M:%S' }
				emptyMessage={ emptyMessage }
				filterParam={ filterParam }
				interactiveLegend={ interactiveLegend }
				interval={ currentInterval }
				isRequesting={ isRequesting }
				itemsLabel={ itemsLabel }
				legendPosition={ legendPosition }
				legendTotals={ legendTotals }
				mode={ mode }
				path={ path }
				query={ query }
				screenReaderFormat={ createDateFormatter(
					formats.screenReaderFormat
				) }
				showHeaderControls={ showHeaderControls }
				title={ selectedChart.label }
				tooltipLabelFormat={ createDateFormatter(
					formats.tooltipLabelFormat
				) }
				tooltipTitle={
					( mode === 'time-comparison' && selectedChart.label ) ||
					null
				}
				tooltipValueFormat={ getTooltipValueFormat(
					selectedChart.type,
					formatAmount
				) }
				chartType={ getChartTypeForQuery( query ) }
				valueType={ selectedChart.type }
				xFormat={ createDateFormatter( formats.xFormat ) }
				x2Format={ createDateFormatter( formats.x2Format ) }
				currency={ getCurrencyConfig() }
			/>
		);
	}

	renderItemComparison() {
		const { isRequesting, primaryData } = this.props;

		if ( primaryData.isError ) {
			return <AnalyticsError />;
		}

		const isChartRequesting = isRequesting || primaryData.isRequesting;
		const chartData = this.getItemChartData();

		return this.renderChart(
			'item-comparison',
			isChartRequesting,
			chartData
		);
	}

	renderTimeComparison() {
		const { isRequesting, primaryData, secondaryData } = this.props;

		if ( ! primaryData || primaryData.isError || secondaryData.isError ) {
			return <AnalyticsError />;
		}

		const isChartRequesting =
			isRequesting ||
			primaryData.isRequesting ||
			secondaryData.isRequesting;
		const chartData = this.getTimeChartData();
		const legendTotals = this.getTimeChartTotals();

		return this.renderChart(
			'time-comparison',
			isChartRequesting,
			chartData,
			legendTotals
		);
	}

	render() {
		const { mode } = this.props;
		if ( mode === 'item-comparison' ) {
			return this.renderItemComparison();
		}
		return this.renderTimeComparison();
	}
}

ReportChart.contextType = CurrencyContext;

ReportChart.propTypes = {
	/**
	 * Filters available for that report.
	 */
	filters: PropTypes.array,
	/**
	 * Whether there is an API call running.
	 */
	isRequesting: PropTypes.bool,
	/**
	 * Label describing the legend items.
	 */
	itemsLabel: PropTypes.string,
	/**
	 * Allows specifying properties different from the `endpoint` that will be used
	 * to limit the items when there is an active search.
	 */
	limitProperties: PropTypes.array,
	/**
	 * `items-comparison` (default) or `time-comparison`, this is used to generate correct
	 * ARIA properties.
	 */
	mode: PropTypes.string,
	/**
	 * Current path
	 */
	path: PropTypes.string.isRequired,
	/**
	 * Primary data to display in the chart.
	 */
	primaryData: PropTypes.object,
	/**
	 * The query string represented in object form.
	 */
	query: PropTypes.object.isRequired,
	/**
	 * Secondary data to display in the chart.
	 */
	secondaryData: PropTypes.object,
	/**
	 * Properties of the selected chart.
	 */
	selectedChart: PropTypes.shape( {
		/**
		 * Key of the selected chart.
		 */
		key: PropTypes.string.isRequired,
		/**
		 * Chart label.
		 */
		label: PropTypes.string.isRequired,
		/**
		 * Order query argument.
		 */
		order: PropTypes.oneOf( [ 'asc', 'desc' ] ),
		/**
		 * Order by query argument.
		 */
		orderby: PropTypes.string,
		/**
		 * Number type for formatting.
		 */
		type: PropTypes.oneOf( [ 'average', 'number', 'currency' ] ).isRequired,
	} ).isRequired,
};

ReportChart.defaultProps = {
	isRequesting: false,
	primaryData: {
		data: {
			intervals: [],
		},
		isError: false,
		isRequesting: false,
	},
	secondaryData: {
		data: {
			intervals: [],
		},
		isError: false,
		isRequesting: false,
	},
};

export default compose(
	withSelect( ( select, props ) => {
		const {
			charts,
			endpoint,
			filters,
			isRequesting,
			limitProperties,
			query,
			advancedFilters,
		} = props;
		const limitBy = limitProperties || [ endpoint ];
		const selectedFilter = getSelectedFilter( filters, query );
		const filterParam = get( selectedFilter, [ 'settings', 'param' ] );
		const chartMode =
			props.mode ||
			getChartMode( selectedFilter, query ) ||
			'time-comparison';
		const { woocommerce_default_date_range: defaultDateRange } = select(
			SETTINGS_STORE_NAME
		).getSetting( 'wc_admin', 'wcAdminSettings' );

		/* eslint @wordpress/no-unused-vars-before-return: "off" */
		const reportStoreSelector = select( REPORTS_STORE_NAME );

		const newProps = {
			mode: chartMode,
			filterParam,
			defaultDateRange,
		};

		if ( isRequesting ) {
			return newProps;
		}

		const hasLimitByParam = limitBy.some(
			( item ) => query[ item ] && query[ item ].length
		);

		if ( query.search && ! hasLimitByParam ) {
			return {
				...newProps,
				emptySearchResults: true,
			};
		}

		const fields = charts && charts.map( ( chart ) => chart.key );

		const primaryData = getReportChartData( {
			endpoint,
			dataType: 'primary',
			query,
			selector: reportStoreSelector,
			limitBy,
			filters,
			advancedFilters,
			defaultDateRange,
			fields,
		} );

		if ( chartMode === 'item-comparison' ) {
			return {
				...newProps,
				primaryData,
			};
		}

		const secondaryData = getReportChartData( {
			endpoint,
			dataType: 'secondary',
			query,
			selector: reportStoreSelector,
			limitBy,
			filters,
			advancedFilters,
			defaultDateRange,
			fields,
		} );

		return {
			...newProps,
			primaryData,
			secondaryData,
		};
	} )
)( ReportChart );
