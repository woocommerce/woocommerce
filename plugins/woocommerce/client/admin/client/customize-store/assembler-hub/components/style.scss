@import "../_mixins.scss";

.woocommerce-customize-store-edit-site-sidebar__screen-wrapper {
	height: 100%;
	overflow-x: auto;
	@include custom-scrollbars-on-hover(transparent, $gray-700);

	// This matches the logo padding
	padding: 0 6px 0 12px;

	// Animation
	animation-duration: 0.14s;
	animation-timing-function: ease-in-out;
	will-change: transform, opacity;

	@media (prefers-reduced-motion: reduce) {
		animation-duration: 0s;
	}

	&.slide-from-left {
		animation-name: slide-from-left;
	}

	&.slide-from-right {
		animation-name: slide-from-right;
	}
}

@keyframes slide-from-right {
	from {
		transform: translateX(50px);
		opacity: 0;
	}
	to {
		transform: none;
		opacity: 1;
	}
}

@keyframes slide-from-left {
	from {
		transform: translateX(-50px);
		opacity: 0;
	}
	to {
		transform: none;
		opacity: 1;
	}
}
