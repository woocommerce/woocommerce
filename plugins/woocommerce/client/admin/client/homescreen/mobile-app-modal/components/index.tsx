export { JetpackInstallationStepper } from './JetpackInstallationStepper';
export {
	useJetpackPluginState,
	JetpackPluginStates,
} from './useJetpackPluginState';
export { useSendMagicLink, SendMagicLinkStates } from './useSendMagicLink';
export { SendMagicLinkButton } from './SendMagicLinkButton';
