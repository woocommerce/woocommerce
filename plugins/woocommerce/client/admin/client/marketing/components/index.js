export { default as ProductIcon } from './product-icon';
export { default as ReadBlogMessage } from './ReadBlogMessage';
export { CollapsibleCard, CardBody, CardDivider } from './CollapsibleCard';
export { PluginCardBody, SmartPluginCardBody } from './PluginCardBody';
export { CardHeaderTitle } from './CardHeaderTitle';
export { CardHeaderDescription } from './CardHeaderDescription';
export { CenteredSpinner } from './CenteredSpinner';
export { CreateNewCampaignModal } from './CreateNewCampaignModal';
