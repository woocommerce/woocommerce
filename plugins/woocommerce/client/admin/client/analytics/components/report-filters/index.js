/**
 * External dependencies
 */
import { Component } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import PropTypes from 'prop-types';
import { omitBy, isUndefined, snakeCase } from 'lodash';
import { withSelect, withDispatch } from '@wordpress/data';
import { STORE_KEY as CES_STORE_KEY } from '@woocommerce/customer-effort-score';
import { ReportFilters as Filters } from '@woocommerce/components';
import { SETTINGS_STORE_NAME } from '@woocommerce/data';
import {
	getCurrentDates,
	getDateParamsFromQuery,
	isoDateFormat,
} from '@woocommerce/date';
import { recordEvent } from '@woocommerce/tracks';
import { CurrencyContext } from '@woocommerce/currency';

/**
 * Internal dependencies
 */
import { LOCALE } from '~/utils/admin-settings';

class ReportFilters extends Component {
	constructor() {
		super();
		this.onDateSelect = this.onDateSelect.bind( this );
		this.onFilterSelect = this.onFilterSelect.bind( this );
		this.onAdvancedFilterAction = this.onAdvancedFilterAction.bind( this );
	}

	onDateSelect( data ) {
		const { report, addCesSurveyForAnalytics } = this.props;
		addCesSurveyForAnalytics();
		recordEvent( 'datepicker_update', {
			report,
			...omitBy( data, isUndefined ),
		} );
	}

	onFilterSelect( data ) {
		const { report, addCesSurveyForAnalytics } = this.props;

		// This event gets triggered in the following cases.
		// 1. Select "Single product" and choose a product.
		// 2. Select "Comparison" or any other filter types.
		// The comparison and other filter types require a user to click
		// a button to execute a query, so this is not a good place to
		// trigger a CES survey for those.
		const triggerCesFor = [
			'single_product',
			'single_category',
			'single_coupon',
			'single_variation',
		];
		const filterName = data.filter || data[ 'filter-variations' ];
		if ( triggerCesFor.includes( filterName ) ) {
			addCesSurveyForAnalytics();
		}

		const eventProperties = {
			report,
			filter: data.filter || 'all',
		};

		if ( data.filter === 'single_product' ) {
			eventProperties.filter_variation =
				data[ 'filter-variations' ] || 'all';
		}

		recordEvent( 'analytics_filter', eventProperties );
	}

	onAdvancedFilterAction( action, data ) {
		const { report, addCesSurveyForAnalytics } = this.props;
		switch ( action ) {
			case 'add':
				recordEvent( 'analytics_filters_add', {
					report,
					filter: data.key,
				} );
				break;
			case 'remove':
				recordEvent( 'analytics_filters_remove', {
					report,
					filter: data.key,
				} );
				break;
			case 'filter':
				const snakeCaseData = Object.keys( data ).reduce(
					( result, property ) => {
						result[ snakeCase( property ) ] = data[ property ];
						return result;
					},
					{}
				);
				addCesSurveyForAnalytics();
				recordEvent( 'analytics_filters_filter', {
					report,
					...snakeCaseData,
				} );
				break;
			case 'clear_all':
				recordEvent( 'analytics_filters_clear_all', { report } );
				break;
			case 'match':
				recordEvent( 'analytics_filters_all_any', {
					report,
					value: data.match,
				} );
				break;
		}
	}

	render() {
		const {
			advancedFilters,
			filters,
			path,
			query,
			showDatePicker,
			defaultDateRange,
		} = this.props;
		const { period, compare, before, after } = getDateParamsFromQuery(
			query,
			defaultDateRange
		);
		const { primary: primaryDate, secondary: secondaryDate } =
			getCurrentDates( query, defaultDateRange );
		const dateQuery = {
			period,
			compare,
			before,
			after,
			primaryDate,
			secondaryDate,
		};
		const Currency = this.context;

		return (
			<Filters
				query={ query }
				siteLocale={ LOCALE.siteLocale }
				currency={ Currency.getCurrencyConfig() }
				path={ path }
				filters={ filters }
				advancedFilters={ advancedFilters }
				showDatePicker={ showDatePicker }
				onDateSelect={ this.onDateSelect }
				onFilterSelect={ this.onFilterSelect }
				onAdvancedFilterAction={ this.onAdvancedFilterAction }
				dateQuery={ dateQuery }
				isoDateFormat={ isoDateFormat }
			/>
		);
	}
}

ReportFilters.contextType = CurrencyContext;

export default compose(
	withSelect( ( select ) => {
		const { woocommerce_default_date_range: defaultDateRange } = select(
			SETTINGS_STORE_NAME
		).getSetting( 'wc_admin', 'wcAdminSettings' );
		return { defaultDateRange };
	} ),
	withDispatch( ( dispatch ) => {
		const { addCesSurveyForAnalytics } = dispatch( CES_STORE_KEY );
		return { addCesSurveyForAnalytics };
	} )
)( ReportFilters );

ReportFilters.propTypes = {
	/**
	 * Config option passed through to `AdvancedFilters`
	 */
	advancedFilters: PropTypes.object,
	/**
	 * Config option passed through to `FilterPicker` - if not used, `FilterPicker` is not displayed.
	 */
	filters: PropTypes.array,
	/**
	 * The `path` parameter supplied by React-Router
	 */
	path: PropTypes.string.isRequired,
	/**
	 * The query string represented in object form
	 */
	query: PropTypes.object,
	/**
	 * Whether the date picker must be shown..
	 */
	showDatePicker: PropTypes.bool,
	/**
	 * The report where filter are placed.
	 */
	report: PropTypes.string.isRequired,
};
