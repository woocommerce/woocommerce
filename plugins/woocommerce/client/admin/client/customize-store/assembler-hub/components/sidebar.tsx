/**
 * External dependencies
 */
import { createContext, useCallback, useState } from '@wordpress/element';
import clsx from 'clsx';

/**
 * Internal dependencies
 */
import './style.scss';

export enum SidebarNavigationAnimationDirection {
	Forward = 'forward',
	Back = 'back',
}

type SidebarNavigationContextType = {
	navigate: ( direction: SidebarNavigationAnimationDirection ) => void;
};

export const SidebarNavigationContext =
	createContext< SidebarNavigationContextType >( {
		navigate: () => void 0,
	} );

export function SidebarContent( { children }: { children: JSX.Element } ) {
	const [ navState, setNavState ] = useState< {
		direction: SidebarNavigationAnimationDirection | null;
	} >( {
		direction: null,
	} );

	const navigate = useCallback(
		( direction: SidebarNavigationAnimationDirection ) => {
			setNavState( { direction } );
		},
		[]
	);

	const wrapperCls = clsx(
		'woocommerce-customize-store-edit-site-sidebar__screen-wrapper',
		{
			'slide-from-left': navState.direction === 'back',
			'slide-from-right': navState.direction === 'forward',
		}
	);

	return (
		<SidebarNavigationContext.Provider value={ { navigate } }>
			<div className={ wrapperCls }>{ children }</div>
		</SidebarNavigationContext.Provider>
	);
}
