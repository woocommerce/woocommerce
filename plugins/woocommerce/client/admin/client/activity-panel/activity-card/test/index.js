/**
 * External dependencies
 */
import { Button } from '@wordpress/components';
import CustomizeIcon from 'gridicons/dist/customize';
import moment from 'moment';
import { render } from '@testing-library/react';
import { createElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { ActivityCard } from '../';

describe( 'ActivityCard', () => {
	test( 'should have correct title', () => {
		const { getByRole } = render(
			<ActivityCard title="Inbox message">
				This card has some content
			</ActivityCard>
		);
		expect(
			getByRole( 'heading', { name: 'Inbox message' } )
		).toBeInTheDocument();
	} );

	test( 'should render a basic card', () => {
		const { container } = render(
			<ActivityCard title="Inbox message">
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render an unread bubble on a card', () => {
		const { container } = render(
			<ActivityCard title="Inbox message" unread>
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a custom icon on a card', () => {
		const { container } = render(
			<ActivityCard title="Inbox message" icon={ <CustomizeIcon /> }>
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a timestamp on a card', () => {
		// We're generating this via moment to ensure it's always "3 days ago".
		const threeDaysAgo = moment().subtract( 3, 'days' ).format();
		const { container } = render(
			<ActivityCard title="Inbox message" date={ threeDaysAgo }>
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'supports a non-date "date" prop on a card', () => {
		// We should be able to provide any string to the date prop.
		const { container } = render(
			<ActivityCard title="Inbox message" date="A long, long time ago">
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render an action on a card', () => {
		const noop = () => {};
		const { container } = render(
			<ActivityCard
				title="Inbox message"
				actions={
					<Button isSecondary onClick={ noop }>
						Action
					</Button>
				}
			>
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render multiple actions on a card', () => {
		const noop = () => {};
		const { container } = render(
			<ActivityCard
				title="Inbox message"
				actions={ [
					<Button key="action1" isPrimary onClick={ noop }>
						Action 1
					</Button>,
					<Button key="action2" isSecondary onClick={ noop }>
						Action 2
					</Button>,
				] }
			>
				This card has some content
			</ActivityCard>
		);
		expect( container ).toMatchSnapshot();
	} );
} );
