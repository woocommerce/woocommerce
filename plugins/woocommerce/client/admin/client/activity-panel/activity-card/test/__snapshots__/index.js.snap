// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard should render a basic card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-notice-outline"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
  </section>
</div>
`;

exports[`ActivityCard should render a custom icon on a card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-customize"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M2 6c0-1.505.78-3.08 2-4 0 .845.69 2 2 2a3 3 0 013 3c0 .386-.079.752-.212 1.091a74.515 74.515 0 012.191 1.808l-2.08 2.08a75.852 75.852 0 01-1.808-2.191A2.977 2.977 0 016 10c-2.21 0-4-1.79-4-4zm12.152 6.848l1.341-1.341A4.446 4.446 0 0017.5 12 4.5 4.5 0 0022 7.5c0-.725-.188-1.401-.493-2.007L18 9l-2-2 3.507-3.507A4.446 4.446 0 0017.5 3 4.5 4.5 0 0013 7.5c0 .725.188 1.401.493 2.007L3 20l2 2 6.848-6.848a68.562 68.562 0 005.977 5.449l1.425 1.149 1.5-1.5-1.149-1.425a68.562 68.562 0 00-5.449-5.977z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
  </section>
</div>
`;

exports[`ActivityCard should render a timestamp on a card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-notice-outline"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
      <span
        class="woocommerce-activity-card__date"
      >
        3 days ago
      </span>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
  </section>
</div>
`;

exports[`ActivityCard should render an action on a card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-notice-outline"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
    <footer
      class="woocommerce-activity-card__actions"
    >
      <button
        class="components-button is-secondary"
        type="button"
      >
        Action
      </button>
    </footer>
  </section>
</div>
`;

exports[`ActivityCard should render an unread bubble on a card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      class="woocommerce-activity-card__unread"
    />
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-notice-outline"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
  </section>
</div>
`;

exports[`ActivityCard should render multiple actions on a card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-notice-outline"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
    <footer
      class="woocommerce-activity-card__actions"
    >
      <button
        class="components-button is-primary"
        type="button"
      >
        Action 1
      </button>
      <button
        class="components-button is-secondary"
        type="button"
      >
        Action 2
      </button>
    </footer>
  </section>
</div>
`;

exports[`ActivityCard supports a non-date "date" prop on a card 1`] = `
<div>
  <section
    class="woocommerce-activity-card"
  >
    <span
      aria-hidden="true"
      class="woocommerce-activity-card__icon"
    >
      <svg
        class="gridicon gridicons-notice-outline"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
          />
        </g>
      </svg>
    </span>
    <header
      class="woocommerce-activity-card__header"
    >
      <h2
        class="woocommerce-activity-card__title"
      >
        Inbox message
      </h2>
      <span
        class="woocommerce-activity-card__date"
      >
        A long, long time ago
      </span>
    </header>
    <div
      class="woocommerce-activity-card__body"
    >
      This card has some content
    </div>
  </section>
</div>
`;
